### DISCUSSION TITLE
A good alternative for unique filtered indexes with SQL Server

### DISCUSSION THREAD

**Discussion by ricardo-valero** [prisma/prisma#9485] [Q&A]
Hi, I was making a site using email and/or phone (so I need them to be nullable) as auth. 

I was using Postgres just fine but then we had to change SQL Server because of reasons 🥲.

And I stumbled on the [Data model limitations](https://www.prisma.io/docs/concepts/database-connectors/sql-server#data-model-limitations) and I know I could create a migration and modify it with [unique filtered indexes](https://stackoverflow.com/a/767702) but I'd have to remove the `@unique`  attribute from the schema.

Is there a better way? Right now I was thinking of making separate tables. But maybe it isn't worth it? Idk.

**Before**
```prisma
model User {
  id             String  @id @default(cuid())
  name           String?
  hashedPassword String?

  email                String?   @unique
  phone                String?   @unique
  emailVerified        Boolean?
  phoneVerified        Boolean?
}
```

**After**
```prisma
model User {
  id             String  @id @default(cuid())
  name           String?
  hashedPassword String?

  email             Email?
  phone             Phone?
}

model Email {
  address    String   @id
  isVerified Boolean?
  user       User     @relation(fields: [userId], references: [id])
  userId     String   @unique
}

model Phone {
  number     String   @id
  isVerified Boolean?
  user       User     @relation(fields: [userId], references: [id])
  userId     String   @unique
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
